src/caffe/solver.cpp

```diff
@@ -409,9 +409,10 @@ Dtype SGDSolver<Dtype>::GetLearningRate() {
   } else if (lr_policy == "step") {
     this->current_step_ = this->iter_ / this->param_.stepsize();
     rate = this->param_.base_lr() *
-        pow(this->param_.gamma(), this->current_step_);
+        caffe_ipow(this->param_.gamma(), this->current_step_);
```

Comment:
it seems to me that caffe_ipow is complete overkill. pow is called once per iteration and takes a fraction of a microsecond.